src/mbed_tools/build/_internal/config/cumulative_data.py

```diff
@@ -41,7 +43,10 @@ def _modify_field(data: CumulativeData, key: str, value: Any) -> None:
     elif modifier == "remove":
         new_value = getattr(data, key) - set(value)
     else:
-        new_value = set(value)
+        if type(value) is str:
+            new_value = str(value)
```

Comment:
I don't understand why we cast to a `str` if `type(x) == str`, seems redundant to me.

Reply:
fixed.